proc/test_thread_rand.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-tests
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	unsigned int time;
+	unsigned int id;
+} threads_self_t;
+
+
+typedef struct {
+	unsigned int cthread;
```

Comment:
`cthread` is misleading, when I change the value below the program will break.

I would prefer to make it `static const int` - then it can be used as a const (so also in structs definition):
```c
static const int cthread = 10;

typedef struct {
	unsigned int fin;
	char stack[cthread][512];
	[...]
}
```

You can also group (sort) struct fields according to their usage (eg. `fin` and `fin_lock`, `stack` and `params`),

Reply:
sorry, my bad, not possible with GCC and pure C (clang actually compiles that even if it's not C standard-compilant). 

We either need to use `#define` or switch to dynamic allocation.

Reply:
I moved `stack` into `thread_self_t` and separated out `time` and `id` into `thread_params_t` structure. Now we can define the amount of threads in one place and still have static allocation. 
Also the split was done to avoid passing thread `stack` as parameter to itself.  

As for `cthread` being misleading: yes its true. Now `cthread` variable is only local for main test run and not global and its value is based on size of `threads` array.